tinytuya/core/XenonDevice.py

```diff
@@ -1,2053 +1,1189 @@
-        alert = "\033[0m\033[91m\033[1m"
-        alertdim = "\033[0m\033[91m\033[2m"
-        cyan = "\033[0m\033[36m"
-        red = "\033[0m\033[31m"
-        yellow = "\033[0m\033[33m"
-    return bold,subbold,normal,dim,alert,alertdim,cyan,red,yellow
-
-
-# Scan function shortcut
-def scan(maxretry=None, color=True, forcescan=False):
-    """Scans your network for Tuya devices with output to stdout"""
-    from . import scanner
-    scanner.scan(scantime=maxretry, color=color, forcescan=forcescan)
-
-
-# Scan function
-def deviceScan(verbose=False, maxretry=None, color=True, poll=True, forcescan=False, byID=False):
-    """Scans your network for Tuya devices and returns dictionary of devices discovered
-        devices = tinytuya.deviceScan(verbose)
-
-    Parameters:
-        verbose = True or False, print formatted output to stdout [Default: False]
-        maxretry = The number of loops to wait to pick up UDP from all devices
-        color = True or False, print output in color [Default: True]
-        poll = True or False, poll dps status for devices if possible
-        forcescan = True or False, force network scan for device IP addresses
-
-    Response:
-        devices = Dictionary of all devices found
-
-    To unpack data, you can do something like this:
-
-        devices = tinytuya.deviceScan()
-        for ip in devices:
-            id = devices[ip]['gwId']
-            key = devices[ip]['productKey']
-            vers = devices[ip]['version']
-            dps = devices[ip]['dps']
-
-    """
-    from . import scanner
-    return scanner.devices(verbose=verbose, scantime=maxretry, color=color, poll=poll, forcescan=forcescan, byID=byID)
+# TinyTuya Module
+# -*- coding: utf-8 -*-
+
+import binascii
+import hmac
+import json
+from hashlib import md5, sha256
+import logging
+import socket
+import struct
+import time
+import sys
+
+from .const import DEVICEFILE, TCPPORT
+from .crypto_helper import AESCipher
+from .error_helper import ERR_CONNECT, ERR_DEVTYPE, ERR_JSON, ERR_KEY_OR_VER, ERR_OFFLINE, ERR_PAYLOAD, error_json
+from .exceptions import DecodeError
+from .message_helper import MessagePayload, TuyaMessage, pack_message, unpack_message, parse_header
+from .command_types import AP_CONFIG, CONTROL, CONTROL_NEW, DP_QUERY, DP_QUERY_NEW, HEART_BEAT, LAN_EXT_STREAM, SESS_KEY_NEG_FINISH, SESS_KEY_NEG_RESP, SESS_KEY_NEG_START, STATUS, UPDATEDPS
+from .header import MESSAGE_HEADER_FMT_55AA, MESSAGE_HEADER_FMT_6699, NO_PROTOCOL_HEADER_CMDS, PREFIX_55AA_BIN, PREFIX_55AA_VALUE, PREFIX_6699_BIN, PREFIX_6699_VALUE, PROTOCOL_3x_HEADER, PROTOCOL_VERSION_BYTES_31, SUFFIX_BIN
+
+log = logging.getLogger(__name__)
+
+# Python 2 Support
+IS_PY2 = sys.version_info[0] == 2
+
+# FIXME: The place where this is used looks like it might have meant `self.version`, but I'm not really sure yet
+#        this is copy/pasted instead of moved because it feels like a bug to me
+version_tuple = (1, 15, 1)
+version = __version__ = "%d.%d.%d" % version_tuple
```

Comment:
This global should not be in this file. It refers to the library version and exists in core.py. The 'self.version' or local version parameters in the functions refer to the Tuya device versions.

Reply:
@jasonacox 

So later in that file (321 in my PR, 948 of core.py in master), there is: 
```python
                        # FIXME: check if we really meant the global `version` here or not
                        self.version_str = "v" + str(version)
                        self.version_bytes = str(version).encode('latin1')
                        self.version_header = self.version_bytes + PROTOCOL_3x_HEADER
                        self.payload_dict = None
```

Which I assume meant to say `new_version` or `self.version` (which are the same at that point), but accidentally uses the global `version` instead.

Actually, it might have been copied from `set_version`, where `version` is an argument.